elements/pl-template/pl-template.py

```diff
@@ -0,0 +1,108 @@
+    required_attribs = ["file-name"]
+    optional_attribs = ["subdirectory", "parent-directory", "warn-undefined"]
+    pl.check_attribs(element, required_attribs, optional_attribs)
+
+    # Load in entries from data dict. Allows filling templates with entries from data['params'], for example.
+    variable_dict = cast(Dict[str, Any], copy.deepcopy(data))
+    options_dict = data["options"]
+
+    for child in element:
+        if child.tag == "pl-variable":
+            pl.check_attribs(
+                child, ["name"], ["file-name", "subdirectory", "parent-directory"]
+            )
+
+            name = pl.get_string_attrib(child, "name")
+
+            if name in variable_dict:
+                raise ValueError(f'Duplicate pl-template variable name: "{name}"')
+
+            inner_html = pl.inner_html(child)
+            has_template_file = pl.has_attrib(child, "file-name")
+
+            if inner_html and has_template_file:
+                raise ValueError(
+                    f'pl-variable "{name}" must have at most one of file-name or its inner html defined'
+                )
+
+            elif has_template_file:
+                variable_file_name = pl.get_string_attrib(child, "file-name")
+                # Default parent directory and subdirectory to what's defined in the outer template.
+                # TODO maybe change the defaults? Could be confusing, and may not save that much writing on the frontend
+                variable_subdirectory = pl.get_string_attrib(
+                    child, "subdirectory", SUBDIRECTORY_DEFAULT
+                )
+
+                variable_parent_directory_choice = pl.get_enum_attrib(
+                    child,
+                    "parent-directory",
+                    ParentDirectoryEnum,
+                    PARENT_DIRECTORY_CHOICE_DEFAULT,
+                )
+
+                variable_parent_directory = options_dict[
+                    variable_parent_directory_choice.value
+                ]
+
+                file_path = os.path.join(
+                    variable_parent_directory, variable_subdirectory, variable_file_name
+                )
+
+                with open(file_path, "r") as f:
+                    variable_dict[name] = f.read()
+
+            else:
+                variable_dict[name] = inner_html
+
+        elif child.tag is lxml.etree.Comment:
+            continue
+
+        else:
+            raise ValueError(
+                f'Tags inside of pl-template must be pl-variable, not "{child.tag}".'
+            )
+
+    parent_directory_choice = pl.get_enum_attrib(
+        element,
+        "parent-directory",
+        ParentDirectoryEnum,
+        PARENT_DIRECTORY_CHOICE_DEFAULT,
+    )
+    parent_directory = options_dict[parent_directory_choice.value]
+
+    subdirectory = pl.get_string_attrib(element, "subdirectory", SUBDIRECTORY_DEFAULT)
+
+    file_name = pl.get_string_attrib(element, "file-name")
+    warn_undefined = pl.get_string_attrib(
```

Comment:
This should be `get_boolean_attrib` so that that the string value gets correctly converted to `True`/`Fasle`

Reply:
Good catch! I think I was on cruise control when I wrote this

Reply:
Sorry I didn't have time to make it a proper pull request. I notice it just before I had to go to work.

Reply:
Another thing I was thinking about this morning was whether the warnings should be on by default or not. The reason I was wondering this was because of how "defaults" are implemented in Mustache. You'd do something like
```html
Your lowest {{num_dropped_assignments}}{{^num_dropped_assignments}}3{{/num_dropped_assignments}} will be dropped in the course. 
```

This relies on the behavior of missing variables being expanded to nothing. I don't know if enough people will be doing this to warrant the warnings to be defaulted to off or not but I thought I should mention it.

Reply:
I was wondering about this. The above template doesn't necessarily rely on the behavior of missing variables, but will perform equivalently if the variables are present and set to false. However, this might create an extra burden on the developer (having to define useless variables), so maybe it's worth defaulting to false and just having it as an option that can be enabled for debugging typos and stuff.

Reply:
@mfbutner in the last push, I change this option to be a boolean and also added an option to trim whitespace (defaulted to True).

As to the discussion below, I'll investigate later if I have time. I'm not quite sure why that would be happening (I'm not an expert on the renderer). Strange that the multiple choice question seems to run its backend code.